docs/media_repository.md

```diff
@@ -2,29 +2,78 @@
 
 *Synapse implementation-specific details for the media repository*
 
-The media repository is where attachments and avatar photos are stored.
-It stores attachment content and thumbnails for media uploaded by local users.
-It caches attachment content and thumbnails for media uploaded by remote users.
+The media repository
+ * stores avatars, attachments and their thumbnails for media uploaded by local
+   users.
+ * caches avatars, attachments and their thumbnails for media uploaded by remote
+   users.
+ * caches resources and thumbnails used for URL previews.
```

Comment:
It might be worth cross-referencing this to the URL preview doc I wrote: https://github.com/matrix-org/synapse/blob/develop/docs/development/url_previews.md

Reply:
Good shout, let's do that.